Thinktecture.Relay.Server/Repository/LinkRepository.cs

```diff
@@ -460,5 +460,23 @@ public void DeleteAllConnectionsForOrigin(Guid originId)
 				_logger?.Error(ex, "Error during deleting of all active connections");
 			}
 		}
+
+		public async Task<bool> HasActiveConnectionAsync(Guid linkId)
+		{
+			_logger?.Verbose("Checking for active connections");
+
+			try
+			{
+				using (var context = new RelayContext())
+				{
+					return await context.ActiveConnections.AnyAsync(ac => ac.LinkId == linkId);
+				}
+			}
+			catch (Exception ex)
+			{
+				_logger?.Error(ex, "Error during checking for active connections");
+				return false;
```

Comment:
When checking for connections, we should take last activity into account:
`ac.LastActivity + _configuration.ActiveConnectionTimeout <= DateTime.UtcNow` , because otherwise we might find an already timed out connection in the database that isn't usable anymore. We don't really have a good cleanup process for inactive connections. When a connection fails and we don't get the disconnected event from the old SignalR we don't delete the entry, it just stays there (possibly forever, until the corresponding origin restarts).

Would it make sense to set isolation level to Read Uncommitted to avoid locking on the DB when reading connections (as last activity on the connection table is also regularly updated and other requests could write while we read)?
 
In case of an exception I would suggest to return `true` (assume we have a connection, we just don't see it at the moment). Worst case is the request timing out as before, but if we just return false because the DB errored out we should not prevent relaying a request.